toplevel/toploop.mli

```diff
@@ -138,10 +138,25 @@ val print_out_phrase :
 
 val read_interactive_input : (string -> bytes -> int -> int * bool) ref
 
-(* Hooks for initialization *)
+(* Hooks *)
 
 val toplevel_startup_hook : (unit -> unit) ref
 
+type event =
+  | Startup
+  | After_setup
```

Comment:
That seems somewhat unrelated to the rest of the PR... What did I miss?

Why was it added?

Reply:
It looks like it becomes necessary for self-contained toplevels?